.github/actions/mobilecoin-cache-cargo-package/action.yaml

```diff
@@ -0,0 +1,31 @@
+name: Mobilecoin Cargo Package Cache
+description: Standardized rust cargo package cache setup
+
+inputs:
+  cache_buster:
+    description: "string to make cache unique"
+    required: false
+  path:
+    description: "path to mount cache"
+    required: false
+    default: |
+      /opt/cargo/git
+      /opt/cargo/registry/index
+      /opt/cargo/registry/cache
+
+outputs:
+  cache-hit:
+    description: "did we get a cache hit?"
+    value: ${{ steps.rust_artifact_cache.outputs.cache-hit }}
+
+runs:
+  using: composite
+  steps:
+  - name: Cache rust build binaries
+    id: rust_artifact_cache
+    uses: actions/cache@v3
+    with:
+      path: ${{ inputs.path }}
+      # Key is a hash of all the .rs, .proto and Cargo.toml files.
+      # if code changes, invalidate cache and rebuild
+      key: ${{ inputs.cache_buster }}-${{ runner.os }}-${{ hashFiles('**/Cargo.toml') }}-cargo-cache
```

Comment:
I think `.cargo/*` is very brittle, doesn't it contain a git repo (the cargo package registry)? I think its timestamps will be changing constantly

Reply:
We can include `**/Cargo.lock` files here too.  

`.cargo/config` doesn't seem to include any packages. I think this would come in to play with the binary cache  .github/actions/mobilecoin-cache-rust-binaries/action.yaml?

Reply:
The mobilecoin-cache-rust-binaries already includes the `.cargo/config` in the hash generation.